EIPS/eip-6404.md

```diff
@@ -0,0 +1,1000 @@
+        schema = (
+            (big_endian_int, tx.chain_id),
+            (big_endian_int, tx.nonce),
+            (big_endian_int, tx.max_fee_per_gas),
+            (big_endian_int, tx.gas_limit),
+            (binary, tx.to if tx.to is not None else []),
+            (big_endian_int, tx.value),
+            (binary, tx.data),
+            (List([Binary[20, 20], List([Binary[32, 32]])]), [
+                (access_tuple.address, access_tuple.storage_keys) for access_tuple in tx.access_list
+            ]),
+            (big_endian_int, 1 if signed_tx.signature.y_parity else 0),
+            (big_endian_int, signed_tx.signature.r),
+            (big_endian_int, signed_tx.signature.s),
+        )
+        sedes = List([schema for schema, _ in schema])
+        values = [value for _, value in schema]
+
+        return keccak([0x01] + rlp.encode(values, sedes))
+
+    assert len(tx.access_list) == 0
+
+    if hash_version == TX_HASH_VERSION_EIP155:
+        schema = (
+            (big_endian_int, tx.nonce),
+            (big_endian_int, tx.max_fee_per_gas),
+            (big_endian_int, tx.gas_limit),
+            (binary, tx.to if tx.to is not None else []),
+            (big_endian_int, tx.value),
+            (binary, tx.data),
+            (big_endian_int, (1 if signed_tx.signature.y_parity else 0) + tx.chain_id * 2 + 35),
+            (big_endian_int, signed_tx.signature.r),
+            (big_endian_int, signed_tx.signature.s),
+        )
+        sedes = List([schema for schema, _ in schema])
+        values = [value for _, value in schema]
+
+        return keccak(rlp.encode(values, sedes))
+
+    assert tx.chain_id == 0
+
+    if hash_version == TX_HASH_VERSION_LEGACY:
+        schema = (
+            (big_endian_int, tx.nonce),
+            (big_endian_int, tx.max_fee_per_gas),
+            (big_endian_int, tx.gas_limit),
+            (binary, tx.to if tx.to is not None else []),
+            (big_endian_int, tx.value),
+            (binary, tx.data),
+            (big_endian_int, (1 if signed_tx.signature.y_parity else 0) + 27),
+            (big_endian_int, signed_tx.signature.r),
+            (big_endian_int, signed_tx.signature.s),
+        )
+        sedes = List([schema for schema, _ in schema])
+        values = [value for _, value in schema]
+
+        return keccak(rlp.encode(values, sedes))
+
+    assert False
+```
+
+```python
+def get_transaction_signer(signed_tx: SignedTransaction) -> Address:
+    if hash_version >= TX_HASH_VERSION_EIP2930:
+        return ecrecover(
+            get_transaction_sighash(signed_tx.tx, signed_tx.hash_version),
+            1 if signed_tx.signature.y_parity else 0,
+            signed_tx.signature.r,
+            signed_tx.signature.s,
+        )
+
+    if hash_version >= TX_HASH_VERSION_EIP155:
+        return ecrecover(
+            get_transaction_sighash(signed_tx.tx, signed_tx.hash_version),
+            (1 if signed_tx.signature.y_parity else 0) + tx.chain_id * 2 + 35,
```

Comment:
I think ecrecover is supposed to operate on a signature that has the chain id removed?

Reply:
Oh, that makes this function obsolete then. The samples I found about `ecrecover` were using the `v` value, so I thought that I need to reconstruct `v` here. Removing `get_transaction_signer`.

The logic inside `get_transaction_id` is still alright, though?